bazel/envoy_library.bzl

```diff
@@ -20,27 +22,14 @@ load(":envoy_pch.bzl", "envoy_pch_copts", "envoy_pch_deps")
 # SelectorValue values have to match the attribute type. See
 # https://github.com/bazelbuild/bazel/issues/2273.
 def tcmalloc_external_deps(repository):
-    return select({
-        repository + "//bazel:disable_tcmalloc": [],
-        repository + "//bazel:disable_tcmalloc_on_linux_x86_64": [],
-        repository + "//bazel:disable_tcmalloc_on_linux_aarch64": [],
-        repository + "//bazel:debug_tcmalloc": [repository + "//bazel/foreign_cc:gperftools"],
-        repository + "//bazel:debug_tcmalloc_on_linux_x86_64": [repository + "//bazel/foreign_cc:gperftools"],
-        repository + "//bazel:debug_tcmalloc_on_linux_aarch64": [repository + "//bazel/foreign_cc:gperftools"],
-        repository + "//bazel:gperftools_tcmalloc": [repository + "//bazel/foreign_cc:gperftools"],
-        repository + "//bazel:gperftools_tcmalloc_on_linux_x86_64": [repository + "//bazel/foreign_cc:gperftools"],
-        repository + "//bazel:gperftools_tcmalloc_on_linux_aarch64": [repository + "//bazel/foreign_cc:gperftools"],
-        repository + "//bazel:linux_x86_64": [
-            "@com_github_google_tcmalloc//tcmalloc",
-            "@com_github_google_tcmalloc//tcmalloc:profile_marshaler",
-            "@com_github_google_tcmalloc//tcmalloc:malloc_extension",
-        ],
-        repository + "//bazel:linux_aarch64": [
-            "@com_github_google_tcmalloc//tcmalloc",
-            "@com_github_google_tcmalloc//tcmalloc:profile_marshaler",
-            "@com_github_google_tcmalloc//tcmalloc:malloc_extension",
-        ],
-        "//conditions:default": [repository + "//bazel/foreign_cc:gperftools"],
+    _repo = repo_label(repository)
+    return selects.with_or({
```

Comment:
I'm sure someone found `with_or` to be more beneficial, but IMHO it makes the code less readable.